mobile/lib/src/application/sleep_sequence_history/sleep_sequence_history_cubit.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:polydodo/src/domain/sleep_sequence/i_sleep_sequence_repository.dart';
+import 'package:polydodo/src/domain/sleep_sequence/sleep_sequence_stats.dart';
+import 'sleep_sequence_history_state.dart';
+
+class SleepSequenceHistoryCubit extends Cubit<SleepSequenceHistoryState> {
+  final ISleepSequenceRepository _sleepHistoryRepository;
+  final StreamController<String> _selectText =
+      StreamController<String>.broadcast();
+
+  List<SleepSequenceStats> _selectedSequences;
+  bool _selectMode = false;
+
+  SleepSequenceHistoryCubit(this._sleepHistoryRepository)
+      : super(SleepSequenceHistoryInitial()) {
+    loadHistory();
+  }
+
+  void loadHistory() {
+    emit(SleepSequenceHistoryLoaded(
+        _sleepHistoryRepository.getSleepSequences()));
+  }
+
+  void selectSleepSequenceForViewing(SleepSequenceStats sequence) {
+    _sleepHistoryRepository.selectSleepSequence(sequence);
+  }
+
+  void toggleSelectMode() {
+    _selectMode = !_selectMode;
+
+    if (_selectMode) {
+      _enableSelection();
+    } else {
+      _disableSelection();
+    }
+  }
+
+  void _enableSelection() {
+    _selectedSequences = [];
+    _selectText.add('Done');
+    emit(SleepSequenceHistoryEditInProgress(
+        _sleepHistoryRepository.getSleepSequences(), _selectedSequences));
+  }
+
+  void _disableSelection() {
+    _selectedSequences = null;
+    _selectText.add('Select');
+    emit(SleepSequenceHistoryLoaded(
+        _sleepHistoryRepository.getSleepSequences()));
+  }
+
+  void toggleSelectSequenceForDeletion(SleepSequenceStats sequence) {
+    if (_selectedSequences.contains(sequence)) {
+      _selectedSequences.remove(sequence);
+    } else {
+      _selectedSequences.add(sequence);
+    }
+
+    emit(SleepSequenceHistoryEditInProgress(
+        _sleepHistoryRepository.getSleepSequences(), _selectedSequences));
+  }
+
+  void deleteSelected() {
+    _sleepHistoryRepository.deleteSleepSequences(_selectedSequences);
+    _disableSelection();
```

Comment:
Ce code ne va pas set `_selectMode` à false, alors peut-être bouger le changement de `_selectMode` dans enable et disable?

Reply:
hmm I either set _selectMode to false here, or add the set to false inside _disableSelection, but also have to keep it in toggle. Because I can't move the sets inside the enable in disable only because the button that toggles is inside the AppBar, and it is harder to change it's behaviour based on the state of the ui since the state is not directly fed to the AppBar. I can also just remove the _disable section from disableSelected. I put it here cause I felt like it made sense, but you don't really have to.

Reply:
Je ne suis pas sur de suivre. T'as plus besoin de garder le `_selectMode` change dans le toggle puisque si enable et disable le font pour toi et que toggle les appelle, alors c'est chill. What am I missing?

Reply:
Nevermind, apparently I was missing something last night, some brain cells.